### DISCUSSION TITLE
Diagram of the Hubs and Spoke stacks

### DISCUSSION THREAD

**Discussion by Charlesc22** [mozilla/hubs#3767] [General Questions & Discussion] [answered]
Does the Hubs team have a visual representation of the Hubs and Spoke stacks?
We are looking for a diagram that shows all the different code bases used in Hubs and Spoke and how they interact with each other
The purpose is to be able to provide a new developer 1 or 2 diagrams that would quickly give them the architecture  and component involved in Hubs.
Could be a stack diagram or an Architecture diagram, but ideally both. You do have one in the Hubs documentation for the AWS architecture, so would like to see one similar in the "Developer" section of the documentation

https://hubs.mozilla.com/docs/hubs-cloud-aws-architecture.html

**Comment by johnshaughnessy** [comment, marked as answer]
We do not have a diagram like that at this time. As for their interaction, `gltf-component-mappings` in the hubs codebase is probably the place to start. Spoke uploads gltf files with components in them to reticulum. The hubs client downloads these gltf files, then handles the components in them as per the mappings in that file.
